src/line/i-chartBuilder.test.js

```diff
@@ -0,0 +1,400 @@
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    const chartTitle = domTesting.getByLabelText(document, "Chart title")
+
+    const xLabel = domTesting.getByLabelText(document, "X label")
+    const yLabel = domTesting.getByLabelText(document, "Y label")
+    
+    const yValues = domTesting.getAllByLabelText(document, "Y")
+    const yValue = yValues[0]
+
+    const buttons = domTesting.queryAllByRole(document, "button")
+    var generateButton = buttons[0]
+
+    //make sure the correct button is selected
+    expect(generateButton).toHaveTextContent("Generate chart")
+
+    //Popups
+    window.alert = jest.fn();
+    const alertMock = jest.spyOn(window, 'alert');
+
+    const user = userEvent.setup()
+
+    await user.type(chartTitle, "Random Chart Title")
+    await user.type(xLabel, "X Label Name :)")
+    await user.type(yLabel, "Y Label Name")
+    await user.type(yValue, "33")
+
+    await user.click(generateButton)
+    
+    expect(alertMock).toHaveBeenCalledTimes(1)
+})
+
+test('Error message appears for missing Y value', async function(){
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    const chartTitle = domTesting.getByLabelText(document, "Chart title")
+
+    const xLabel = domTesting.getByLabelText(document, "X label")
+    const yLabel = domTesting.getByLabelText(document, "Y label")
+    
+    const xValues = domTesting.getAllByLabelText(document, "X")
+    const xValue = xValues[0]
+
+    const buttons = domTesting.queryAllByRole(document, "button")
+    var generateButton = buttons[0]
+
+    //make sure the correct button is selected
+    expect(generateButton).toHaveTextContent("Generate chart")
+
+    //Popups
+    window.alert = jest.fn();
+    const alertMock = jest.spyOn(window, 'alert');
+
+    const user = userEvent.setup()
+
+    await user.type(chartTitle, "Random Chart Title")
+    await user.type(xLabel, "X Label Name :)")
+    await user.type(yLabel, "Y Label Name :)")
+    await user.type(xValue, "22")
+
+    await user.click(generateButton)
+    
+    expect(alertMock).toHaveBeenCalledTimes(1)
+})
+
+
+//CLEARING CHART DATA
+test('All chart values are cleared when Clear Chart Data is pressed', async function(){
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    const chartTitle = domTesting.getByLabelText(document, "Chart title")
+    const xLabel = domTesting.getByLabelText(document, "X label")
+    const yLabel = domTesting.getByLabelText(document, "Y label")
+
+    const buttons = domTesting.queryAllByRole(document, "button")
+    const clearButton = buttons[1]
+
+    //make sure the correct button is selected
+    expect(clearButton).toHaveTextContent("Clear chart data")
+
+    const user = userEvent.setup()
+
+    await user.type(chartTitle, "Random Chart Title")
+    await user.type(xLabel, "X Label Name :)")
+    await user.type(yLabel, "Y Label Name :)")
```

Comment:
Based on the test name, I would expect this test to ensure the color input and X/Y input pairs are cleared when the "Clear chart data" button is pressed. Could you please add that?